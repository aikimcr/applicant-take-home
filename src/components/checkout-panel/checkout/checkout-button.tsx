import React from 'react';
import { Button } from '../../common';

interface CheckoutButtonProps {
    onClick: () => void;
}

export const CheckoutButton: React.FC<CheckoutButtonProps> = ({ onClick }): React.ReactElement => {
    const buttonText = 'Prizeout Gift Card';
    return (
        <>
            <Button
                ariaLabel="Prizeout your gift card"
                color={`primary`}
                onClick={onClick}
                size="medium"
                text={buttonText}
                type="submit"
            />
        </>
    );
};

export default CheckoutButton;
