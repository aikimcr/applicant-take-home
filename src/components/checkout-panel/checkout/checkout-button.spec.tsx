import { render, screen } from '../../../testing/test-utils';
import React from 'react';
import '@testing-library/jest-dom';

import { CheckoutButton } from './checkout-button';

describe('Test CheckoutButtone', () => {
    test('Component matches snapshot', () => {
        const clickHandler = jest.fn();
        const { asFragment } = render(<CheckoutButton onClick={clickHandler} />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('submit giftcard on click', () => {
        const clickHandler = jest.fn();
        render(<CheckoutButton onClick={clickHandler} />);

        const button = screen.getByText('Prizeout Gift Card');
        button.click();

        expect(clickHandler).toBeCalledTimes(1);
    });
});
