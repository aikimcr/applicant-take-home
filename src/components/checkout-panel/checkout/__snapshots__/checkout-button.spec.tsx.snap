// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CheckoutButtone Component matches snapshot 1`] = `
<DocumentFragment>
  <button
    aria-label="Prizeout your gift card"
    class="button button--primary button--medium"
    data-testid="button"
    title="Prizeout Gift Card"
    type="submit"
  >
    <span>
      Prizeout Gift Card
    </span>
  </button>
</DocumentFragment>
`;
